spec/spec.md

```diff
@@ -422,10 +425,13 @@ where:
 
 - `auth_token`: an opaque string representing a unique identifying token issued by the wallet endpoint to the dapp endpoint. The format and contents are an implementation detail of the wallet endpoint. The dapp endpoint can use this on future connections to reauthorize access to [privileged methods](#privileged-methods).
 - `accounts`: one or more value objects that represent the accounts to which this auth token corresponds. These objects hold the following properties:
-  - `address`: a base64-encoded address for this account
+  - `address`: the address for this account. The format of this string will depend on the chain, and is specified by the `addressFormat` field
+  - `address_format`: (optional) the format of the `address`. Defaults to `base64` for backwards compatibility with previous versions of this spec. 
```

Comment:
Previously, the intent of the `address` field was conflated with the public key (since they are the same for Solana). We selected base64 encoding because this wasn't intended to be a human-visible field. Rather, it was present for preparing transactions. If the dApp wanted to display the address to the user, it was responsible for formatting that value itself. Because of that, this looks like a breaking change to me. The default is base64, so everything will probably work on day 1, but what if a wallet decides to start returning this as, e.g., base58?

I think we should accept that our terminology of `address` was broken and note that. Keep the legacy behavior, where `address` is actually the public key, and add a new field, e.g. `display_address` and `display_address_format`.

WDYT?

Reply:
yeah I like that. `address` is effectively a public key and probably should have been called that from the beginning. it made sense for 1.0 being solana specific, we have to eat that now. 

I think we should use `display_*` as you suggest and also add a `public_key` param as an alias for `address`. in the future we could remove `address` in favor of `public_key`